### PR TITLE
fix: use updateRoundStep to track consensus state changes

### PR DESCRIPTION
**Pull request by cmwaters**
We're currently using `newStep` to indicate whether to write a new trace, however this actually doesn't track when the round and step change happens and thus misses certain steps (like NewRound). This PR fixes it

### CODE REVIEW

**Review by rootulp — APPROVED** [review]

**Review by rach-id — APPROVED** [review]

File: consensus/state.go [hunk 1]
```diff
@@ -1518,6 +1515,7 @@ func (cs *State) enterPrecommitWait(height int64, round int32) {
 	defer func() {
 		// Done enterPrecommitWait:
 		cs.TriggeredTimeoutPrecommit = true
+		cs.updateRoundStep(round, cstypes.RoundStepPrecommitWait)
 		cs.newStep()
```
Comment (staheri14) [inline consensus/state.go:1519]: We also have one called in[ newStep()](https://github.com/celestiaorg/celestia-core/blob/602544164e68fca964ae2dd16e09eadc7ba7c5ed/consensus/state.go#L711), wouldn't that cause duplicates?

Comment (staheri14) [reply consensus/state.go:1519]: I misread the code, the change is fine!

**Review by staheri14 — APPROVED** [review]

File: consensus/state.go [hunk 2]
```diff
@@ -538,6 +538,7 @@ func (cs *State) updateRoundStep(round int32, step cstypes.RoundStepType) {
 		}
 		if cs.Step != step {
 			cs.metrics.MarkStep(cs.Step)
+			schema.WriteRoundState(cs.traceClient, cs.Height, round, uint8(step))
```
Comment (staheri14) [inline consensus/state.go:541]: I guess my previous question applies here: Isn't `updateRoundStep` always followed by `newStep`? Then we already trace changes in `newStep` which means we will have duplicates?

Comment (cmwaters) [reply consensus/state.go:541]: This is not always the case. For instance I notice the NewRound steo was never triggered and also the PrecommitWait step

Comment (cmwaters) [reply consensus/state.go:541]: Oh and I removed the tracing from `newStep`

**Pull request merged**